src/view/terminal/termion_terminal.rs

```diff
@@ -275,6 +276,10 @@ impl Terminal for TermionTerminal {
             guard.take();
         }
 
+        // Flush the terminal before suspending to cause the switch from the
+        // alternate screen to main screen to properly restore the terminal.
+        let _ = stdout().flush();
```

Comment:
@christoph-heiss can we call this on `self.output` instead? I realize they may fundamentally do the same thing, since `stdout` is a singleton, but it would be more consistent with other write operations, and in the unlikely event that `AlternateScreen` does anything but delegate, that logic would still get called.

Can you please try that on line 273 when we have the output mutex value? If it works without issue I'd prefer that. :innocent:

Reply:
I just tried it and it doesn't seem to work. 

After looking at termion's source, the switch between main/alternate screen is done using a special escape sequence, which `AlternateScreen` screen writes to `stdout` on `drop()`. And that won't be recognized until we do a `flush()` on `stdout` (or a println!() for that matter).
And the flush on `self.output` would be done before dropping it, so that actually makes sense why it doesn't work I guess.

Reply:
That's helpful; thanks for trying it out!